File: day2/part1.py
# part 1

fp = open("input.txt")
par=[0,0]
while True:
    b = fp.readline()
    if b == '':
        break
    #print(b)

    pares=[False,False]

    for i in range(0,len(b)):
        z=1
        for j in range(0,len(b)):
            if i == j:
                continue
            if b[i] == b[j]:
                z=z+1
        if z==2:
            pares[0]=True
        if z==3:
            pares[1]=True
    if pares[0]==True:
        par[0]=par[0]+1
    if pares[1]==True:
        par[1]=par[1]+1

print(par)
print(par[0]*par[1])
    

File: day2/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}
func analiza(str string, num int) bool {

	for i := 0; i < len(str); i++ {
		if strings.Count(str, string(str[i])) == num {
			return true
		}
	}
	return false
}

func main() {
	par := [2]int{0, 0}
	file, err := os.Open("input.txt")
	check(err)
	defer file.Close()

	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		str := scanner.Text()

		if analiza(str, 2) {
			par[0]++
		}

		if analiza(str, 3) {
			par[1]++
		}

	}
	fmt.Println(par)
	fmt.Println(par[0] * par[1])
}
